api/lib/redisHelper.mjs

```diff
@@ -26,17 +27,24 @@ export function getClient(databaseIndex = 0) {
  * Gets the value of a specified key in the database.
  * @param {string} key the key of the entry to get.
  * @param {number} databaseIndex the index the entry is stored in.
+ * @throws {NotFoundError} if the key is not in the database.
  * @returns {object} the entry's information.
  */
 export async function getEntry(key, databaseIndex = 0) {
     const client = getClient(databaseIndex);
     await client.connect();
 
-    const res = await client.get(key);
-    // TODO: Throw error if the key does not exist in the db.
-
-    await client.quit();
-
+    let res;
+    try {
+        res = await client.get(key);
+        // Check if the key exists, otherwise throw a custom error
+        if (res === null) {
+            console.error('Error while fetching key "%s": %s', key, error.message);
+            throw new NotFoundError(`Key "${key}" not found in database ${databaseIndex}`);
```

Comment:
I really like your logic here!  My only comment is what I mentioned at the top: the redis client should not be throwing http errors (that's what the api does hence the separation in folders).  Instead, you should make a copy of the `MisformedKeyError` and amend it to reflect the error data for your error.